File: App/src/Adaptive.ReactiveTrader.Client/Models/ITrade.cs
using System;

namespace Adaptive.ReactiveTrader.Client.Models
{
    public interface ITrade
    {
        string CurrencyPair { get; }
        Direction Direction { get; }
        long Notional { get; }
        decimal SpotRate { get; }
        TradeStatus TradeStatus { get; }
        DateTime TradeDate { get; }
        long TradeId { get; }
        string TraderName { get; }
        DateTime ValueDate { get; }
    }
}

File: App/src/Adaptive.ReactiveTrader.Client.JS/Models/Execution/ITrade.ts
interface ITrade {
    currencyPair: string;
    direction: Direction;
    notional: number;
    spotRate: number;
    tradeStatus: TradeStatus;
    tradeDate: Date;
    tradeId: number;
    traderName:string;
    valueDate:Date;
    dealtCurrency: string;
} 